substrate/frame/revive/src/exec.rs

```diff
@@ -1642,7 +1651,14 @@ where
 		if block_number < self.block_number.saturating_sub(256u32.into()) {
 			return None;
 		}
-		Some(System::<T>::block_hash(&block_number).into())
+
+		// Fallback to the system block hash for older blocks
+		// 256 entries should suffice for all use cases, this mostly ensures
+		// our benchmarks are passing.
+		match crate::Pallet::<T>::eth_block_hash_from_number(block_number.into()) {
+			Some(hash) => Some(hash),
```

Comment:
Possibly the hash needs to be decoded similarly to how is has been changed in [the gas mapping PR](https://github.com/paritytech/polkadot-sdk/pull/9803/files#diff-ccc183489bcbd5a91f42d7c462c72f4f5993b23dec8869d113ec324aae66692dR1650) (@athei?)